simphony/testing/abc_check_engine.py

```diff
@@ -212,6 +215,7 @@ def test_dataset_rename(self):
         # we should be able to access using the new "bar" name
         ds_bar = engine.get_dataset("bar")
         self.assertEqual(ds_bar.name, "bar")
+        engine.remove_dataset("bar")
```

Comment:
Ah, okay.  I see your point now.  I prefer the first solution you proposed using the if-statement. Something like:
```
if self.number_datasets_used_in_testing > 1:
   # and we should be able to use the no-longer used
   # "foo" name when adding another dataset 	# "foo" name when adding another dataset
   ds = engine.add_dataset(self.create_dataset(name='foo')) 	
   ds = engine.add_dataset(self.create_dataset(name='foo'))
```